### PR TITLE
 add more linters 

### PR DESCRIPTION
**Pull request by fmoral2**
#### Proposed Changes ####

- adding more important linters: 

* If you want to test it, just use the new golanci yaml in your local main 

Linters link: 
  

- - funlen https://golangci-lint.run/usage/linters/#funlen
- - gocritic - https://golangci-lint.run/usage/linters/#gocritic
- - godot - https://golangci-lint.run/usage/linters/#godot
- - perfsprint - https://golangci-lint.run/usage/linters/#perfsprint
- - usestdlibvars - https://golangci-lint.run/usage/linters/#usestdlibvars
- - whitespace - https://golangci-lint.run/usage/linters/#whitespace
 
- Which returned me some important errors   related to code security, performance and maintainability , which are all fixed .

#### Testing ####

Checklist:
1. If your PR changes anything on or related to Jenkins, run it pointing to your branch to make sure it's okay.

2. Verify code lint; we should not have errors.

3. Update the documentation if needed.

4. Update makefile and docker run if adding new tests.

5. Run your tests at least 4 times with all configurations needed and possible.

6. If needed test with different os types.

### CODE REVIEW

File: pkg/assert/pod.go [hunk 1]
```diff
@@ -57,25 +56,7 @@ func checkReadyFields() types.GomegaMatcher {
 	}, BeTrue())
 }
 
-// PodAssertStatus custom assertion that asserts that pod status is completed or in some cases
-// apply pods can have an error status
-func PodAssertStatus() PodAssertFunc {
```
Comment (fmoral2) [inline pkg/assert/pod.go:62]: this is removed due to we are already doing this validation on 

https://github.com/rancher/distros-test-framework/blob/8329d5ac4184f0378e973304a0f8164988cb1bd0/pkg/testcase/pod.go#L41

**Review by rancher-max — APPROVED** [review]

**Comment by mdrahman-suse** [comment]
Needs rebasing and linter error fixes

**Comment by fmoral2** [comment]
will fix all tomorrow.

**Comment by fmoral2** [comment]
Tested on k3s and rke2

- dual stack
- validate cluster
- all upgrades
- cert
- secrets
- cluster reset

File: pkg/testcase/dualstack.go [hunk 2]
```diff
@@ -127,14 +128,37 @@ func TestIPFamiliesDualStack(deleteWorkload bool) {
...
-		"k3s_issue_10053_pod1.yaml", "k3s_issue_10053_pod2.yaml")
-	Expect(err).NotTo(HaveOccurred(), "failed to deploy initial manifests")
+	serverPodIP, err := getPodIP(cluster)
+	Expect(err).NotTo(HaveOccurred(), "setupPods failed")
 
-	// Ensure the pods are running and retrieve the correct pod IP
-	var serverPodIP string
+	err = validatePodConnectivity(cluster, serverPodIP, deleteWorkload)
+	Expect(err).NotTo(HaveOccurred(), "validatePodConnectivity failed")
+}
```
Comment (rancher-max) [inline pkg/testcase/dualstack.go:136]: Rearranging this makes it a bit confusing to me what all is happening. I think it just comes down to function naming at the end of the day. Would it work/pass linting if we did this instead?

```go
func TestIngressWithPodRestartAndNetPol(cluster *factory.Cluster, deleteWorkload bool) {
	// Deploy server and client pods
	err := shared.ManageWorkload("apply", "k3s_issue_10053_ns.yaml",
		"k3s_issue_10053_pod1.yaml", "k3s_issue_10053_pod2.yaml")
	Expect(err).NotTo(HaveOccurred(), "failed to deploy initial manifests")
	filters := map[string]string{
		"namespace": "test-k3s-issue-10053",
	}
	serverPodIP := getServerPodIP(cluster, filters)

	// Ensure connectivity from client pod to server pod
	assert.ValidateIntraNSPodConnectivity("test-k3s-issue-10053", "client", serverPodIP, "Hostname: server")

	// Deploy network policy that explicitly allows access to the server pod
	err = shared.ManageWorkload("apply", "k3s_issue_10053_netpol.yaml")
	Expect(err).NotTo(HaveOccurred(), "whoami pod failed to deploy")

	// Ensure connectivity from client pod to server pod BEFORE restarting the server
	assert.ValidateIntraNSPodConnectivity("test-k3s-issue-10053", "client", serverPodIP, "Hostname: server")

	// Redeploy server pod and ensure it is up and running again. Retrieve its new IP.
	err = shared.ManageWorkload("delete", "k3s_issue_10053_pod1.yaml")
	Expect(err).NotTo(HaveOccurred(), "whoami pod failed to delete")
	err = shared.ManageWorkload("apply", "k3s_issue_10053_pod1.yaml")
	Expect(err).NotTo(HaveOccurred(), "whoami pod failed to redeploy")
	serverPodIP = getServerPodIP(cluster, filters)

	// Ensure connectivity from client pod to server pod AFTER restarting the server
	assert.ValidateIntraNSPodConnectivity("test-k3s-issue-10053", "client", serverPodIP, "Hostname: server")

	if deleteWorkload {
		err = shared.ManageWorkload("delete", "k3s_issue_10053_ns.yaml")
		Expect(err).NotTo(HaveOccurred())
	}
}

func getServerPodIP(cluster *factory.Cluster, filters map[string]string) string {
	var podIP string
	Eventually(func(g Gomega) {
		pods, poderr := shared.GetPodsFiltered(filters)
		g.Expect(poderr).NotTo(HaveOccurred())
		g.Expect(pods).NotTo(BeEmpty())

		for _, pod := range pods {
			processPodStatus(cluster, g, pod,
				assert.PodAssertRestart(),
				assert.PodAssertReady())

			if pod.Name == "server" {
				podIP = pod.IP
			}
		}
	}, "120s", "5s").Should(Succeed())
	return podIP
}
```

File: pkg/testcase/ingressdns.go [hunk 3]
```diff
@@ -128,27 +128,27 @@ func validateIngressRoute(publicIp string) {
 	Eventually(func(g Gomega) {
 		pods, getErr := shared.GetPodsFiltered(filters)
 		g.Expect(getErr).NotTo(HaveOccurred(), getErr)
-		for _, pod := range pods {
-			g.Expect(pod.IP).NotTo(Equal("<none>"))
-			g.Expect(pod.IP).NotTo(BeEmpty())
-			if pod.IP != "<none>" && pod.IP != "" {
+		for i := range pods {
```
Comment (rancher-max) [inline pkg/testcase/ingressdns.go:131]: I personally prefer the for loops using `for _, pod` 😕

File: pkg/testcase/ingressdns.go [hunk 4]
```diff
@@ -90,13 +90,13 @@ func TestIngressRoute(cluster *factory.Cluster, applyWorkload, deleteWorkload bo
 
 		replacer := strings.NewReplacer("$YOURDNS", publicIp, "$APIVERSION", apiVersion)
 		newContent := replacer.Replace(string(content))
-		errWrite := os.WriteFile(newFilePath, []byte(newContent), 0644)
+		errWrite := os.WriteFile(newFilePath, []byte(newContent), 0o644)
```
[the remark this answers is not in the document]
Comment (rancher-max) [reply pkg/testcase/ingressdns.go:93]: never mind on this... I understand the reasoning as it is an octal number and should be distinguished as such.

File: pkg/testcase/dualstack.go (hunk 2 above)
Comment (fmoral2) [reply pkg/testcase/dualstack.go:136]: will try

File: pkg/testcase/ingressdns.go (hunk 3 above)
Comment (fmoral2) [reply pkg/testcase/ingressdns.go:131]: i prefer too but this one is a performance linter warning 
```
pkg/testcase/ingressdns.go:130:3: rangeValCopy: each iteration copies 160 bytes (consider pointers or indexing) (gocritic)
                for _, pod := range pods {
                ^
```

Comment (rancher-max) [reply pkg/testcase/ingressdns.go:131]: Dangit gocritic! I'll accept it I guess

File: pkg/testcase/dualstack.go (hunk 2 above)
Comment (fmoral2) [reply pkg/testcase/dualstack.go]: it works

Comment (fmoral2) [reply pkg/testcase/dualstack.go]: @rancher-max

Comment (fmoral2) [reply pkg/testcase/dualstack.go]: https://github.com/rancher/distros-test-framework/pull/118/commits/ec4cac03d0ddebb2aa16174322490fb7401e8fe9

**Review by rancher-max — APPROVED** [review]

**Comment by fmoral2** [comment]
@rancher-max  needed to revert the `TestIngressWithPodRestartAndNetPol` and just removed some lines, due to an error i was getting not being able to reach out the pod again, separating the getserverPODip 
```
 Trying to assert with:
 Connecting to 10.42.1.10 (10.42.1.10:80)
 wget: can't connect to remote host (10.42.1.10): Connection refused
 command terminated with exit code 1
 
```

**Review by rancher-max — APPROVED** [review]

**Review by ShylajaDevadiga — APPROVED** [review]

**Pull request merged**